File: polocloud-components/component-terminal/src/main/java/dev/httpmarco/polocloud/component/terminal/command/CommandArgument.java
package dev.httpmarco.polocloud.component.terminal.command;


import java.util.List;

public abstract class CommandArgument<T> {

    private final String key;

    public CommandArgument(String key) {
        this.key = key;
    }

    public List<String> defaultArgs(CommandContext context) {
        return List.of();
    }

    // if one argument must be special type
    public boolean predication(String rawInput) {
        return !(rawInput.startsWith("<") && rawInput.endsWith(">"));
    }

    public String wrongReason() {
        return"";
    }

    public abstract T buildResult(String input);

    public String key() {
        return key;
    }


}

File: agent/src/main/java/dev/httpmarco/polocloud/agent/runtime/local/terminal/commands/CommandArgument.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.commands

import dev.httpmarco.polocloud.agent.i18n

abstract class CommandArgument<T>(open val key: String) {

    open fun defaultArgs(context: CommandContext): MutableList<String> {
        return mutableListOf()
    }

    // if one argument must be a special type
    open fun predication(rawInput: String): Boolean {
        return !(rawInput.startsWith("<") && rawInput.endsWith(">"))
    }

    open fun wrongReason(): String {
        return i18n.get("agent.terminal.command.reason.wrong")
    }

    abstract fun buildResult(input: String, context: CommandContext): T
}